api/src/task/notifications/closing_date_notification.py

```diff
@@ -0,0 +1,185 @@
+import logging
+from datetime import timedelta
+from uuid import UUID
+
+import botocore.client
+from sqlalchemy import and_, exists, select
+
+from src.adapters import db
+from src.db.models.opportunity_models import Opportunity, OpportunitySummary
+from src.db.models.user_models import UserOpportunityNotificationLog, UserSavedOpportunity
+from src.task.notifications.base_notification import BaseNotification
+from src.task.notifications.constants import EmailData, NotificationReasons
+from src.util import datetime_util
+
+logger = logging.getLogger(__name__)
+
+CONTACT_INFO = (
+    "mailto:support@grants.gov\n"
+    "1-800-518-4726\n"
+    "24 hours a day, 7 days a week\n"
+    "Closed on federal holidays"
+)
+
+
+class ClosingDateNotification(BaseNotification):
+
+    def __init__(
+        self,
+        db_session: db.Session,
+        app_id: str,
+        pinpoint_client: botocore.client.BaseClient | None = None,
+        frontend_base_url: str | None = None,
+    ):
+        super().__init__(db_session)
+        self.app_id = app_id
+        self.pinpoint_client = pinpoint_client
+        self.frontend_base_url = frontend_base_url
+        self.collected_data: dict | None = None
```

Comment:
Since the pinpoint stuff is shared between every notification task, can we move the `self.x = ...` bit up to that class for app ID and pinpoint client?